File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil {
        return nil
    }
    current := head
    for head.Next != nil {
        if head.Val == head.Next.Val {
            head.Next = head.Next.Next
        }
        if head.Next != nil && head.Val != head.Next.Val {
            head = head.Next
        }
    }
    return current
}

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if (head === null) {
        return null;
    }
    let current = head;
    while (head.next !== null) {
        if (head.val === head.next.val) {
            head.next = head.next.next;
        }
        if (head.next !== null && head.val !== head.next.val) {
            head = head.next;
        }
    }
    return current;
};